service_discovery/README.md

```diff
@@ -0,0 +1,196 @@
+# Service Discovery
+
+- [Introduction](#introduction)
+- [High-level Design](#high-level-design)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Building](#building)
+  - [Running the Tests](#running-the-tests)
+- [Running the Service](#running-the-service)
+  - [Sample Applications](#sample-applications)
+- [Developing with Service Discovery](#developing-with-service-discovery)
+- [Trademarks](#trademarks)
+
+## Introduction
+
+This Service Discovery system for Eclipse Chariott aims to simplify in-vehicle application development by abstracting out where services are running, so that applications that want to leverage the resources and capabilities of another application can discover the location and metadata necessary to communicate. This is a new version of Service Discovery which is decoupled from the intent broker, and it is currently under development. If you would like to use the existing version of Chariott with intent brokering, please refer to the [top-level Readme](./../README.md) for details.
+
+The [Getting Started](#getting-started) section shows how to get started and run the Service Discovery system locally.
+
+The [Developing with Service Discovery](#developing-with-service-discovery) shows what is necessary to start developing applications that use Service Discovery.
+
+## High-level Design
+
+The service discovery system consists of 3 parts:
+
+1. The Service Registry
+2. A provider
+3. A consumer
+
+The Chariott Service Registry stores enough metadata about a service for a consumer to be able to communicate directly with the provider. A service is uniquely identified by its namespace, name, and version. Namespace is a logical grouping of services. An example in the repo is "sdv.samples", which is the namespace for all of the sample services. Today, there can only be one service registered with the same namespace, name, version combination, and any attempt to register again will fail. See the ServiceMetadata type in the [service registry proto](./proto/core/v1/service_registry.proto) for more detailed information and examples of the metadata required for each service.
+
+Providers and consumers are applications, or any software components. An application can take on the role or "provider", "consumer", or both. A provider is an application which registers itself and gets added to the service registry, whereas a consumer is an appplication which searches for applications in the service registry, using one of the read operations on the registry (i.e. Discover to retrieve a single service).
```

Comment:
Please change the second sentence to “An application can take on the role of a “provider”, “consumer”, or both.”